modules/actions/snapshots/fill-missing-snapshots-v2.ts

```diff
@@ -0,0 +1,110 @@
+                },
+            },
+        },
+        where: {
+            vaultVersion: 2,
+            chain,
+        },
+    });
+
+    // Return list of missing timestamps for a pool
+    const filledIn: string[] = [];
+    for (const pool of pools) {
+        const poolId = pool.id;
+        const missingTimestamps = await prisma.$queryRawUnsafe<{ day: number }[]>(
+            `
+            WITH pool_snapshots AS (
+                SELECT * FROM "PrismaPoolSnapshot"
+                WHERE chain = '${chain}'
+                AND "poolId" = '${poolId}'
+            ), date_range AS (
+                SELECT generate_series(MIN(DATE_TRUNC('day', to_timestamp(timestamp)))::date, MAX(DATE_TRUNC('day', to_timestamp(timestamp)))::date, '1 day'::interval) AS day
+                FROM pool_snapshots
+            )
+            SELECT 
+                EXTRACT(EPOCH from dr.day)::int as day
+            FROM date_range dr
+            LEFT JOIN pool_snapshots e ON dr.day = DATE_TRUNC('day', to_timestamp(timestamp))::date
+            WHERE e.id IS NULL
+            ORDER BY dr.day;
+        `,
+        );
+
+        if (missingTimestamps.length === 0) {
+            continue;
+        }
+
+        console.log(`Pool ${poolId} has ${missingTimestamps.length} missing snapshots.`);
+
+        for (const missingTimestamp of missingTimestamps) {
+            const timestamp = missingTimestamp.day;
+            const previousSnapshot = await prisma.prismaPoolSnapshot.findFirst({
+                where: {
+                    chain: chain,
+                    poolId: poolId,
+                    timestamp: timestamp - 86400,
+                },
+            });
+
+            if (!previousSnapshot) {
+                console.log(`Could not find previous snapshot for pool ${poolId} on day ${timestamp}.`);
+                continue;
+            }
+
+            // Get prices for that day
+            const prices = (
+                await prisma.prismaTokenPrice.findMany({
+                    where: {
+                        chain,
+                        timestamp,
+                    },
+                    select: {
+                        tokenAddress: true,
+                        price: true,
+                    },
+                })
+            ).reduce((acc, { tokenAddress, price }) => {
+                acc[tokenAddress] = price;
+                return acc;
+            }, {} as Record<string, number>);
+
+            const tokens =
+                pools
+                    .find((p) => p.id === poolId)
+                    ?.tokens.reduce((acc, t) => {
+                        acc[t.index] = t.address;
+                        return acc;
+                    }, [] as string[]) ?? [];
+
+            const data = snapshotsV2Transformer(poolId, tokens, timestamp, chain, prices, previousSnapshot);
+
+            if (!data) {
+                console.log(`Could not fill in missing snapshot for pool ${poolId} on day ${timestamp}.`);
```

Comment:
I guess this could happen if we dont have prices for a missing snapshot. What happens with the sync if we still have gaps in the snapshots?